src/general/Mutations.cs

```diff
@@ -170,96 +170,132 @@ private void MutateBehaviour(Species parent, Species mutated)
+                }
+                else
+                {
+                    // Duplicate an existing organelle, but only if there are any organelles where that is legal
+                    var organellesThatCanBeDuplicated =
+                        parentOrganelles.Organelles.Where(organelle => !organelle.Definition.Unique).ToList();
+                    if (organellesThatCanBeDuplicated.Any())
                     {
-                        AddNewOrganelle(mutatedOrganelles, GetRandomOrganelle(isBacteria));
+                        AddNewOrganelle(mutatedOrganelles,
+                            organellesThatCanBeDuplicated.Random(random).Definition);
                     }
                     else
                     {
-                        // Duplicate an existing organelle, but only if there are any organelles where that is legal
-                        var organellesThatCanBeDuplicated =
-                            parentOrganelles.Organelles.Where(organelle => !organelle.Definition.Unique).ToList();
-                        if (organellesThatCanBeDuplicated.Any())
-                        {
-                            AddNewOrganelle(mutatedOrganelles,
-                                organellesThatCanBeDuplicated.Random(random).Definition);
-                        }
-                        else
-                        {
-                            AddNewOrganelle(mutatedOrganelles, GetRandomOrganelle(isBacteria));
-                        }
+                        AddNewOrganelle(mutatedOrganelles, GetRandomOrganelle(isBacteria));
                     }
                 }
             }
+        }
 
-            if (isBacteria)
+        if (isBacteria)
+        {
+            if (random.Next(0.0f, 1.0f) <= Constants.MUTATION_BACTERIA_TO_EUKARYOTE)
             {
-                if (random.Next(0.0f, 1.0f) <= Constants.MUTATION_BACTERIA_TO_EUKARYOTE)
-                {
-                    AddNewOrganelle(mutatedOrganelles, nucleus);
-                }
+                AddNewOrganelle(mutatedOrganelles, nucleus);
             }
+        }
+
+        // Disallow creating empty species as that throws an exception when trying to spawn
+        if (mutatedOrganelles.Count < 1)
+        {
+            // Add the first parent species organelle
+            AddNewOrganelle(mutatedOrganelles, parentOrganelles[0].Definition);
 
-            // Disallow creating empty species as that throws an exception when trying to spawn
+            // If still empty, copy the first organelle of the parent
             if (mutatedOrganelles.Count < 1)
+                mutatedOrganelles.Add((OrganelleTemplate)parentOrganelles[0].Clone());
+        }
+
+        var islandHexes = mutatedOrganelles.GetIslandHexes();
+
+        // Attach islands
+        while (islandHexes.Count > 0)
+        {
+            var mainHexes = mutatedOrganelles.ComputeHexCache().Except(islandHexes);
```

Comment:
Does this work correctly if there are multiple islands on different sides of the microbe? What if those are exactly symmetrical and the average distance is zero?

That's basically my only code concern. This PR basically just needs a lot of testing to make sure it never fails.